client/cli/lib/start.ts

```diff
@@ -1,33 +1,77 @@
-import * as liveServer from 'live-server';
-import bodyParser from 'body-parser';
+import cors from 'cors';
+import express from 'express';
+import { resolve } from 'path';
 
 import build, { Options } from './build';
-import { validateDependencies } from './utils';
+import { getComponentDirectories, validateDependencies } from './utils';
 
 export const DEFAULT_PORT = 8383;
 
-export default async ({ port = DEFAULT_PORT }, buildOptions?: Options): Promise<any> => {
-  const { main, peerDependencies } = await build(buildOptions);
-
-  return liveServer.start({
-    port,
-    root: './dist',
-    file: main,
-    open: false,
-    cors: true,
-    middleware: [
-      bodyParser.json(),
-      (req, res, next) => {
-        if (req.method == 'GET') {
-          res.setHeader('Access-Control-Expose-Headers', 'ETag');
-        }
-
-        if (req.method === 'POST') {
-          validateDependencies(req.body, peerDependencies);
-        }
-
-        next();
+export type StartOptions = {
+  port?: number;
+  workspaces?: string[];
+};
+
+export default async (
+  logger,
+  { port = DEFAULT_PORT, workspaces }: StartOptions,
+  buildOptions?: Options
+): Promise<any> => {
+  const dir = buildOptions && buildOptions.dir;
+
+  const dirs = getComponentDirectories(dir, workspaces);
+  const builds = await Promise.all(
+    dirs.map(async dir => {
+      const result = await build({ ...buildOptions, dir });
+      return { ...result, dir };
+    })
+  );
+
+  const index: Record<string, any> = builds.reduce(
+    (acc, { name, ...result }) => ({
+      ...acc,
+      [name]: result
+    }),
+    {}
+  );
+
+  const app = express();
+
+  app.set('etag', 'strong');
+
+  app.use(express.json(), cors());
+
+  app
+    .route('/:component')
+    .all((req, res, next) => {
```

Comment:
this check can be skipped if you create the routes in a `forEach` loop on the index...

Reply:
It also adds the component to the request

Reply:
I saw that but if you work in the closure of a foreach you get the component's name from the parameters so you don't need to add the name... 
```
index.forEach(nameOfComponents => app.route(/nameOfComponents).get((req, res) => {
      res.setHeader('Access-Control-Expose-Headers', 'ETag');
      const component = nameOfComponent;
      res.sendFile(resolve(process.cwd(), component.dir, 'dist', component.main));
    }));
```

Removes validation code... Does it make sense?

Reply:
but then you don't get the logger warning if you're trying to get a non existing component